haskellPackages: update stackage and hackage

sternenseemann opened the pull request:
### This Merge

This PR is the regular merge of the `haskell-updates` branch into `master`.

This branch is being continually built and tested by hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. You may be able to find an up-to-date Hydra build report at [cdepillabout/nix-haskell-updates-status](https://github.com/cdepillabout/nix-haskell-updates-status).

We roughly aim to merge these `haskell-updates` PRs at least once every two weeks. See the @NixOS/haskell [team calendar](https://cloud.maralorn.de/apps/calendar/p/H6migHmKX7xHoTFa) for who is currently in charge of this branch.

### haskellPackages Workflow Summary

Our workflow is currently described in [`pkgs/development/haskell-modules/HACKING.md`](https://github.com/NixOS/nixpkgs/blob/haskell-updates/pkgs/development/haskell-modules/HACKING.md).

The short version is this:
* We regularly update the Stackage and Hackage pins on `haskell-updates` (normally at the beginning of a merge window).
* The community fixes builds of Haskell packages on that branch.
* We aim at at least one merge of `haskell-updates` into `master` every two weeks.
* We only do the merge if the [`mergeable`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/mergeable) job is succeeding on hydra.
* If a [`maintained`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/maintained) package is still broken at the time of merge, we will only merge if the maintainer has been pinged 7 days in advance. (If you care about a Haskell package, become a maintainer!)

More information about Haskell packages in nixpkgs can be found [in the nixpkgs manual](https://nixos.org/manual/nixpkgs/unstable/#haskell).

---

This is the follow-up to #351154. Come to [#haskell:nixos.org](https://matrix.to/#/#haskell:nixos.org) if you have any questions.

sternenseemann committed:
haskell.compiler.ghc94{5,6}: remove at 9.4.5 and 9.4.6, respectively

Both are affected by <https://gitlab.haskell.org/ghc/ghc/-/issues/23762>
which is now cropping up on our CI (for Cabal 3.14). I don't really see
why we absolutely have to keep them, so it seems just better overall to
use this as an excuse to remove them (after NixOS 24.11 branch off).

sternenseemann committed:
all-cabal-hashes: 2024-10-25T11:10:52Z -> 2024-11-07T13:09:31Z

This commit has been generated by maintainers/scripts/haskell/update-hackage.sh

sternenseemann committed:
haskellPackages: stackage LTS 22.39 -> LTS 22.40

This commit has been generated by maintainers/scripts/haskell/update-stackage.sh

sternenseemann committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

sternenseemann committed:
haskellPackages.haskell-language-server: refactor override

K900 commented:
I think we have to make a call here, and have haskell-updates target staging, not master. We have a variety of things that depend on haskell-adjacent packages (like pandoc) that are _not_ part of the haskell-updates jobset and can't realistically be without it becoming a second staging, and every haskell-updates merge causes an unhealthy amount of rebuilds for master - e.g. https://github.com/NixOS/nixpkgs/pull/351154 rebuilds Qt, among other things.

maralorn commented:
I am confused. Is there something special about this specific PR? Or is this a general opinion?

Also, pandoc is part of our tested set. I am not aware of any significant number of rebuilds caused by merging haskell-updates into master. If they are that's probably a bug in our workflow.

maralorn commented:
For which plattforms did that trigger the rebuilds? Is that maybe because we got stripped of the darwin builders? Or was this bad luck and the merge coincided with a merge of staging-next?

We really have a workflow specifically designed to prevent rebuilds on master on merge.

maralorn commented:
How many rebuilds are that?
I wasn't aware of this problem.

github-actions[bot] committed:
Merge master into haskell-updates

vcunat commented:
[OfBorg's estimate](https://gist.github.com/GrahamcOfBorg/ba3f884e397079971a8ee0361e6372a0) from that iteration was around 51k rebuilds total (all 4 platforms).  The haskell-updates jobset prepared about 14k, which is partially because we deactivated its darwin builds some time ago.

sternenseemann commented:
So it seems that most (if not all) of the rebuilds are caused by the use of `shellcheck` in `writeShellApplication`. I was aware of this, but perceived it as an API for external users, so I never realized it was used in nixpkgs/NixOS to this extent.

I guess we could do (some of) the following things:

- Clamp down on the use of `writeShellApplication` in nixpkgs. Executing `shellcheck` can be done in `passthru.tests` for nixpkgs, so I don't really see why it needs to be used in this way, it currently is. Even without the unfortunate situation with `haskell-updates`, this would reduce rebuild time for a lot of packages probably.
- Send `haskell-updates` to `staging`. This of course will significantly increase the time to deliver updates to packages. The 2 weeks is of course just a soft goal that's more a function of how maintenance duties have been organized. In practice, it depends on availability of maintainer and build time—sometimes it goes quicker, sometimes slower. However, downstream users are occasionally dissatisfied even with the speed we can deliver updates at the moment. A possible setup would be the following: `haskell-updates` is based on `master` so that changes can be tested with less disruptions and then merged into `staging` (or an early `staging-next`). We'd probably need to tie the automatic merges to the ones for `staging`, so that haskell-updates doesn't get ahead of staging. We'd also start being more liberal with merging low rebuild fixes into `master`.
- Extend the normal jobsets to include Haskell specific jobs and use them instead of the custom one, making `haskell-updates` a second staging. Probably infeasible as long as there are as many jobs rebuilt by Haskell changes.

sternenseemann committed:
haskell.compiler.ghcHEAD: disable --hyperlinked-source on aarch64

Disable enableHyperlinkedSource for GHC >= 9.11 to keep GHC under the
3GB Hydra output limit. Note that we use a patch as opposed to the new
extraArgs interfaces for the reasons given in the comments added in this
commit.

sternenseemann commented:
This iteration won't reach 24.11. We'll probably want to backport some stuff, in particular anything relating to HLS and GHC 9.8.3.

sternenseemann committed:
haskell.compiler.ghcHEAD: bootstrap using GHC 9.10

GHC 9.6 can be used in a pinch, but not to build e.g. a cross compiler.

sternenseemann committed:
haskell.packages.ghcHEAD: start compiler config for GHC 9.14

sternenseemann committed:
haskellPackages: adjust to updates on Hackage

In the case of stack, our package set has catched up with its demands
partially.

danielrolls committed:
haskellPackages.aws-spend-summary add maintainer

sternenseemann committed:
Merge branch staging-next into haskell-updates

sternenseemann committed:
haskellPackages.ghcide: apply patch for GHC 9.8.3 compat

danielrolls committed:
aws-spend-summary: init

maralorn committed:
aws-spend-summary: init (#354805)

mpscholten commented:
now that haskell language server is fixed, can we make GHC 9.8.3 the default for ghc98?

sternenseemann commented:
Discussing at #354789. We should do some more testing, probably.

alexfmpe committed:
haskellPackages.reflex: remove patches that were upstreamed

sternenseemann committed:
Merge branch staging into haskell-updates

alexfmpe committed:
haskellPackages.dependent-sum-template: build with ghc 9.10

alexfmpe committed:
haskellPackages.dependent-sum-template: remove jailbreak on 9.8

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann commented:
We'll have to wait for the next `staging-next` iteration to restart this iteration (i.e. for the current staging becoming staging-next). This is because I accidentally merged in `staging` instead of `staging-next`. Unfortunately, I don't think there is a way to revert this merge without making it impossible to properly merge in `staging-next` or `master` until that merge is reverted (see <https://git.github.io/htmldocs/howto/revert-a-faulty-merge.html>).

github-actions[bot] committed:
Merge master into haskell-updates

maralorn committed:
haskellPackages.reflex: remove patches that were upstreamed (#355022)

maralorn committed:
haskellPackages.dependent-sum-template: build with ghc 9.10 (#355052)

alexfmpe committed:
haskellPackages: fix test suites on multiple stream libraries

alexfmpe committed:
haskellPackages.testcontainers: fix build

alexfmpe committed:
haskellPackages.beam-postgres: fix build

danielrolls committed:
aws-spend-summary: add shell completions

alexfmpe committed:
haskellPackages: cleanup overrides after linked pulls made it to hackage

maralorn committed:
haskellPackages: fix test suites on multiple stream libraries (#356314)

maralorn committed:
haskellPackages.beam-postgres: fix build (#356378)

maralorn committed:
aws-spend-summary: add shell completions (#356409)

maralorn committed:
haskellPackages: cleanup overrides after linked pulls made it to hackage (#356650)

sternenseemann committed:
Merge branch staging into haskell-updates

Contains the following interesting changes for haskell-updates:

- #356918 (which we can revert now)
- #320148 (which is a set rebuild)

sternenseemann committed:
Revert "git-annex: work around corrupted store paths of dependencies"

This reverts commit 5e1144c572a133b129d004275e3afde53e248fce.

Reason for revert: Due to unrelated set rebuilds on
staging/haskell-updates compared to master, we get freshly built store
paths of these packages anyways.

sternenseemann commented:
This PR will resolve #356741.

github-actions[bot] committed:
Merge master into haskell-updates

alexfmpe committed:
haskellPackages.hailgun: fix build

alexfmpe committed:
haskellPackages.servant-auth-server: unbreak

maralorn committed:
haskellPackages.hailgun: fix build (#358192)

maralorn committed:
haskellPackages.servant-auth-server: unbreak (#358193)

alexfmpe committed:
haskellPackages.http-reverse-proxy: disable tests on darwin

maralorn committed:
haskellPackages.http-reverse-proxy: disable tests on darwin (#358486)

sternenseemann committed:
Merge branch staging into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
Merge branch staging into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
Merge branch staging-next into haskell-updates

sternenseemann commented:
`git log origin/staging-next..origin/haskell-updates` are just Haskell-related changes now, so we should be good to re-enable the Hydra jobset soon. Then we should be able to work on release stabilisation as normal, but also “offline” compilation on the branch should at least no longer require bootstrapping nixpkgs yourself…

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
Merge branch staging-next into haskell-updates

sternenseemann committed:
Merge branch staging-next into haskell-updates

sternenseemann committed:
Merge branch staging-next into haskell-updates

sternenseemann commented:
Solved merge conflict due to reformat of `haskell-packages.nix`.

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
Merge branch staging-next into haskell-updates

vcunat commented:
So, about a Hydra jobset.  I haven't noticed what are the plans around platforms, but I assume it's best to do the first eval just with x86_64-linux?  (and add something later?)

sternenseemann commented:
> So, about a Hydra jobset. I haven't noticed any plans around platforms, but I assume it's best to do the first eval just with x86_64-linux? (and add something later?)

Sounds good! *-darwin is sometimes quite problematic, aarch64-linux is unproblematic, but I think builds take longer in general (although this may have changed).

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskellPackages.yi-language: drop obsolete patch

sternenseemann committed:
Merge branch staging-next into haskell-updates

sternenseemann committed:
Merge branch staging-next into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann commented:
### [haskell-updates build report from hydra](https://hydra.nixos.org/jobset/nixpkgs/haskell-updates)
*evaluation [1810539](https://hydra.nixos.org/eval/1810539) of nixpkgs commit [873b5ff](https://github.com/NixOS/nixpkgs/commits/873b5ff14545ee502918a74993cfb77b474e2805) as of 2024-12-21 01:55 UTC*

🔴 **Branch not mergeable**
  * `mergeable` jobset failed.
  * `maintained` jobset failed.

#### Build summary

 | Platform | Failed ❌ | DependencyFailed ❗ | Unfinished ⏳ | Success ✅ | 
 | --- | --- | --- | --- | --- | 
 | [x86_64-linux 🐧](https://hydra.nixos.org/eval/1810539?filter=.x86_64-linux) | 22 | 28 | 1 | 6644 | 
#### Maintained Linux packages with build failure
- [ ] [[🐧❌]](https://hydra.nixos.org/build/282180234) [haskellPackages.espial](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.espial) @dalpd
- [ ] [jacinda](https://hydra.nixos.org/eval/1810539?filter=jacinda) @sternenseemann
  - [[🐧❌]](https://hydra.nixos.org/build/282177603) [toplevel](https://hydra.nixos.org/eval/1810539?filter=jacinda)
  - [[🐧❌]](https://hydra.nixos.org/build/282180888) [haskellPackages](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.jacinda)
- [ ] [stack](https://hydra.nixos.org/eval/1810539?filter=stack) @cdepillabout
  - [[🐧❌]](https://hydra.nixos.org/build/282178984) [toplevel](https://hydra.nixos.org/eval/1810539?filter=stack)
  - [[🐧❌]](https://hydra.nixos.org/build/282175021) [haskellPackages](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.stack)
#### Maintained Linux packages with failed dependency
- [ ] [cachix](https://hydra.nixos.org/eval/1810539?filter=cachix) @domenkozar
  - [[🐧❗]](https://hydra.nixos.org/build/282216443) [toplevel](https://hydra.nixos.org/eval/1810539?filter=cachix)
  - [[🐧❗]](https://hydra.nixos.org/build/282216399) [haskellPackages](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.cachix)
- [ ] [ghc](https://hydra.nixos.org/eval/1810539?filter=ghc) @cdepillabout @guibou @maralorn @ncfavier @sternenseemann
  - [[🐧✅]](https://hydra.nixos.org/build/282176013) [haskellPackages](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.ghc)
  - [[🐧❗]](https://hydra.nixos.org/build/282216434) [pkgsCross.ghcjs.haskell.packages.ghc98](https://hydra.nixos.org/eval/1810539?filter=pkgsCross.ghcjs.haskell.packages.ghc98.ghc)
  - [[🐧❗]](https://hydra.nixos.org/build/282216469) [pkgsCross.ghcjs.haskell.packages.ghcHEAD](https://hydra.nixos.org/eval/1810539?filter=pkgsCross.ghcjs.haskell.packages.ghcHEAD.ghc)
  - [[🐧❗]](https://hydra.nixos.org/build/282216412) [pkgsCross.ghcjs.haskellPackages](https://hydra.nixos.org/eval/1810539?filter=pkgsCross.ghcjs.haskellPackages.ghc)
- [ ] [git-annex](https://hydra.nixos.org/eval/1810539?filter=git-annex) @peti @roosemberth
  - [[🐧❗]](https://hydra.nixos.org/build/282177295) [toplevel](https://hydra.nixos.org/eval/1810539?filter=git-annex)
  - [[🐧❗]](https://hydra.nixos.org/build/282177586) [haskellPackages](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.git-annex)
- [ ] [haskell-language-server](https://hydra.nixos.org/eval/1810539?filter=haskell-language-server) @maralorn
  - [[🐧✅]](https://hydra.nixos.org/build/282175346) [toplevel](https://hydra.nixos.org/eval/1810539?filter=haskell-language-server)
  - [[🐧✅]](https://hydra.nixos.org/build/282176775) [haskell.packages.ghc9101](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc9101.haskell-language-server)
  - [[🐧✅]](https://hydra.nixos.org/build/282179723) [haskell.packages.ghc925](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc925.haskell-language-server)
  - [[🐧✅]](https://hydra.nixos.org/build/282177708) [haskell.packages.ghc926](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc926.haskell-language-server)
  - [[🐧✅]](https://hydra.nixos.org/build/282177099) [haskell.packages.ghc927](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc927.haskell-language-server)
  - [[🐧✅]](https://hydra.nixos.org/build/282179921) [haskell.packages.ghc928](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc928.haskell-language-server)
  - [[🐧✅]](https://hydra.nixos.org/build/282180385) [haskell.packages.ghc947](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc947.haskell-language-server)
  - [[🐧✅]](https://hydra.nixos.org/build/282180856) [haskell.packages.ghc948](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc948.haskell-language-server)
  - [[🐧✅]](https://hydra.nixos.org/build/282179799) [haskell.packages.ghc963](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc963.haskell-language-server)
  - [[🐧✅]](https://hydra.nixos.org/build/282174521) [haskell.packages.ghc964](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc964.haskell-language-server)
  - [[🐧✅]](https://hydra.nixos.org/build/282179197) [haskell.packages.ghc965](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc965.haskell-language-server)
  - [[🐧✅]](https://hydra.nixos.org/build/282181028) [haskell.packages.ghc966](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc966.haskell-language-server)
  - [[🐧✅]](https://hydra.nixos.org/build/282174741) [haskell.packages.ghc981](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc981.haskell-language-server)
  - [[🐧✅]](https://hydra.nixos.org/build/282178231) [haskell.packages.ghc982](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc982.haskell-language-server)
  - [[🐧✅]](https://hydra.nixos.org/build/282179198) [haskell.packages.ghc983](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc983.haskell-language-server)
  - [[🐧❗]](https://hydra.nixos.org/build/282177103) [haskell.packages.ghc984](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc984.haskell-language-server)
  - [[🐧✅]](https://hydra.nixos.org/build/282180311) [haskellPackages](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.haskell-language-server)
- [ ] [[🐧❗]](https://hydra.nixos.org/build/282216410) [hci](https://hydra.nixos.org/eval/1810539?filter=hci) @roberth
- [ ] [[🐧❗]](https://hydra.nixos.org/build/282216427) [haskellPackages.hercules-ci-cli](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.hercules-ci-cli) @roberth
- [ ] [[🐧❗]](https://hydra.nixos.org/build/282257610) [maintained](https://hydra.nixos.org/eval/1810539?filter=maintained) @cdepillabout @maralorn @ncfavier @sternenseemann
- [ ] [[🐧❗]](https://hydra.nixos.org/build/282216441) [mergeable](https://hydra.nixos.org/eval/1810539?filter=mergeable) @cdepillabout @maralorn @ncfavier @sternenseemann
#### Unmaintained packages with build failure
<details><summary>38 job(s) </summary>

- [ ] [ghc-lib-parser](https://hydra.nixos.org/eval/1810539?filter=ghc-lib-parser)  ⤴️ 19 | 70
  - [[🐧✅]](https://hydra.nixos.org/build/282177083) [haskell.packages.ghc8107](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc8107.ghc-lib-parser)
  - [[🐧✅]](https://hydra.nixos.org/build/282174764) [haskell.packages.ghc902](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc902.ghc-lib-parser)
  - [[🐧✅]](https://hydra.nixos.org/build/282180656) [haskell.packages.ghc925](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc925.ghc-lib-parser)
  - [[🐧✅]](https://hydra.nixos.org/build/282174747) [haskell.packages.ghc926](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc926.ghc-lib-parser)
  - [[🐧✅]](https://hydra.nixos.org/build/282179267) [haskell.packages.ghc927](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc927.ghc-lib-parser)
  - [[🐧✅]](https://hydra.nixos.org/build/282179832) [haskell.packages.ghc928](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc928.ghc-lib-parser)
  - [[🐧✅]](https://hydra.nixos.org/build/282180100) [haskell.packages.ghc947](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc947.ghc-lib-parser)
  - [[🐧✅]](https://hydra.nixos.org/build/282178972) [haskell.packages.ghc948](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc948.ghc-lib-parser)
  - [[🐧✅]](https://hydra.nixos.org/build/282178094) [haskell.packages.ghc963](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc963.ghc-lib-parser)
  - [[🐧✅]](https://hydra.nixos.org/build/282176137) [haskell.packages.ghc964](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc964.ghc-lib-parser)
  - [[🐧✅]](https://hydra.nixos.org/build/282177819) [haskell.packages.ghc965](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc965.ghc-lib-parser)
  - [[🐧✅]](https://hydra.nixos.org/build/282178467) [haskell.packages.ghc966](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc966.ghc-lib-parser)
  - [[🐧✅]](https://hydra.nixos.org/build/282180763) [haskell.packages.ghc981](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc981.ghc-lib-parser)
  - [[🐧✅]](https://hydra.nixos.org/build/282175458) [haskell.packages.ghc982](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc982.ghc-lib-parser)
  - [[🐧✅]](https://hydra.nixos.org/build/282176187) [haskell.packages.ghc983](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc983.ghc-lib-parser)
  - [[🐧❌]](https://hydra.nixos.org/build/282175099) [haskell.packages.ghc984](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc984.ghc-lib-parser)
  - [[🐧✅]](https://hydra.nixos.org/build/282175506) [haskellPackages](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.ghc-lib-parser)
- [ ] [[🐧❌]](https://hydra.nixos.org/build/282177055) [haskellPackages.termbox-bindings-c](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.termbox-bindings-c)  ⤴️ 4 | 4
- [ ] [[🐧❌]](https://hydra.nixos.org/build/282176056) [haskellPackages.lzma-conduit](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.lzma-conduit)  ⤴️ 3 | 7
- [ ] [[🐧❌]](https://hydra.nixos.org/build/282178103) [haskellPackages.lbfgs](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.lbfgs)  ⤴️ 2 | 3
- [ ] [[🐧❌]](https://hydra.nixos.org/build/282179482) [haskellPackages.HsSyck](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.HsSyck)  ⤴️ 1 | 10
- [ ] [[🐧❌]](https://hydra.nixos.org/build/282179513) [haskellPackages.libxml-sax](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.libxml-sax)  ⤴️ 0 | 21
- [ ] [darcs](https://hydra.nixos.org/eval/1810539?filter=darcs)  ⤴️ 0 | 1
  - [[🐧❌]](https://hydra.nixos.org/build/282180674) [toplevel](https://hydra.nixos.org/eval/1810539?filter=darcs)
  - [[🐧❌]](https://hydra.nixos.org/build/282177626) [haskellPackages](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.darcs)
- [ ] [[🐧❌]](https://hydra.nixos.org/build/282180893) [haskellPackages.dhscanner-ast](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.dhscanner-ast)  ⤴️ 0 | 1
- [ ] [[🐧❌]](https://hydra.nixos.org/build/282178852) [haskellPackages.bindings-libcddb](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.bindings-libcddb) 
- [ ] [[🐧❌]](https://hydra.nixos.org/build/282179090) [haskellPackages.diagrams-input](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.diagrams-input) 
- [ ] [[🐧❌]](https://hydra.nixos.org/build/282176202) [haskellPackages.gpu-vulkan-core](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.gpu-vulkan-core) 
- [ ] [[🐧❌]](https://hydra.nixos.org/build/282181177) [haskellPackages.hdf5-lite](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.hdf5-lite) 
- [ ] [[🐧❌]](https://hydra.nixos.org/build/282180795) [haskellPackages.miniterion](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.miniterion) 
- [ ] [[🐧❌]](https://hydra.nixos.org/build/282175703) [haskellPackages.posix-timer](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.posix-timer) 
- [ ] [[🐧❌]](https://hydra.nixos.org/build/282180175) [haskellPackages.powerqueue-distributed](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.powerqueue-distributed) 
- [ ] [[🐧❌]](https://hydra.nixos.org/build/282178131) [haskellPackages.relocant](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.relocant) 
- [ ] [[🐧❌]](https://hydra.nixos.org/build/282178661) [haskellPackages.sandwich-contexts](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.sandwich-contexts) 
- [ ] [[🐧❌]](https://hydra.nixos.org/build/282175434) [haskellPackages.servant-hateoas](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.servant-hateoas) 
- [ ] [[🐧❌]](https://hydra.nixos.org/build/282175333) [haskellPackages.xtea](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.xtea) 
</details>

#### Unmaintained packages with failed dependency
<details><summary>88 job(s) </summary>

- [ ] [microlens](https://hydra.nixos.org/eval/1810539?filter=microlens)  ⤴️ 168 | 600
  - [[🐧✅]](https://hydra.nixos.org/build/282178509) [haskellPackages](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.microlens)
  - [[🐧❗]](https://hydra.nixos.org/build/282216417) [pkgsCross.ghcjs.haskell.packages.ghc98](https://hydra.nixos.org/eval/1810539?filter=pkgsCross.ghcjs.haskell.packages.ghc98.microlens)
  - [[🐧❗]](https://hydra.nixos.org/build/282216449) [pkgsCross.ghcjs.haskell.packages.ghcHEAD](https://hydra.nixos.org/eval/1810539?filter=pkgsCross.ghcjs.haskell.packages.ghcHEAD.microlens)
  - [[🐧❗]](https://hydra.nixos.org/build/282216465) [pkgsCross.ghcjs.haskellPackages](https://hydra.nixos.org/eval/1810539?filter=pkgsCross.ghcjs.haskellPackages.microlens)
- [ ] [ghc-lib-parser-ex](https://hydra.nixos.org/eval/1810539?filter=ghc-lib-parser-ex)  ⤴️ 13 | 44
  - [[🐧✅]](https://hydra.nixos.org/build/282177946) [haskell.packages.ghc8107](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc8107.ghc-lib-parser-ex)
  - [[🐧✅]](https://hydra.nixos.org/build/282174790) [haskell.packages.ghc902](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc902.ghc-lib-parser-ex)
  - [[🐧✅]](https://hydra.nixos.org/build/282176921) [haskell.packages.ghc925](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc925.ghc-lib-parser-ex)
  - [[🐧✅]](https://hydra.nixos.org/build/282177852) [haskell.packages.ghc926](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc926.ghc-lib-parser-ex)
  - [[🐧✅]](https://hydra.nixos.org/build/282179914) [haskell.packages.ghc927](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc927.ghc-lib-parser-ex)
  - [[🐧✅]](https://hydra.nixos.org/build/282176015) [haskell.packages.ghc928](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc928.ghc-lib-parser-ex)
  - [[🐧✅]](https://hydra.nixos.org/build/282179613) [haskell.packages.ghc947](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc947.ghc-lib-parser-ex)
  - [[🐧✅]](https://hydra.nixos.org/build/282179220) [haskell.packages.ghc948](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc948.ghc-lib-parser-ex)
  - [[🐧✅]](https://hydra.nixos.org/build/282176562) [haskell.packages.ghc963](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc963.ghc-lib-parser-ex)
  - [[🐧✅]](https://hydra.nixos.org/build/282180973) [haskell.packages.ghc964](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc964.ghc-lib-parser-ex)
  - [[🐧✅]](https://hydra.nixos.org/build/282181005) [haskell.packages.ghc965](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc965.ghc-lib-parser-ex)
  - [[🐧✅]](https://hydra.nixos.org/build/282175683) [haskell.packages.ghc966](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc966.ghc-lib-parser-ex)
  - [[🐧✅]](https://hydra.nixos.org/build/282180661) [haskell.packages.ghc981](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc981.ghc-lib-parser-ex)
  - [[🐧✅]](https://hydra.nixos.org/build/282177943) [haskell.packages.ghc982](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc982.ghc-lib-parser-ex)
  - [[🐧✅]](https://hydra.nixos.org/build/282180529) [haskell.packages.ghc983](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc983.ghc-lib-parser-ex)
  - [[🐧❗]](https://hydra.nixos.org/build/282176983) [haskell.packages.ghc984](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc984.ghc-lib-parser-ex)
  - [[🐧✅]](https://hydra.nixos.org/build/282181390) [haskellPackages](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.ghc-lib-parser-ex)
- [ ] [ihaskell](https://hydra.nixos.org/eval/1810539?filter=ihaskell)  ⤴️ 7 | 17
  - [[🐧❗]](https://hydra.nixos.org/build/282216468) [toplevel](https://hydra.nixos.org/eval/1810539?filter=ihaskell)
  - [[🐧✅]](https://hydra.nixos.org/build/282177225) [haskellPackages](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.ihaskell)
- [ ] [[🐧❗]](https://hydra.nixos.org/build/282175667) [haskellPackages.termbox-bindings-hs](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.termbox-bindings-hs)  ⤴️ 3 | 3
- [ ] [[🐧❗]](https://hydra.nixos.org/build/282174880) [haskellPackages.termbox](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.termbox)  ⤴️ 2 | 2
- [ ] [hoogle](https://hydra.nixos.org/eval/1810539?filter=hoogle)  ⤴️ 1 | 5
  - [[🐧✅]](https://hydra.nixos.org/build/282176971) [haskell.packages.ghc8107](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc8107.hoogle)
  - [[🐧✅]](https://hydra.nixos.org/build/282179097) [haskell.packages.ghc902](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc902.hoogle)
  - [[🐧❗]](https://hydra.nixos.org/build/282180233) [haskell.packages.ghc9101](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc9101.hoogle)
  - [[🐧✅]](https://hydra.nixos.org/build/282177962) [haskell.packages.ghc925](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc925.hoogle)
  - [[🐧✅]](https://hydra.nixos.org/build/282174366) [haskell.packages.ghc926](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc926.hoogle)
  - [[🐧✅]](https://hydra.nixos.org/build/282176349) [haskell.packages.ghc927](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc927.hoogle)
  - [[🐧✅]](https://hydra.nixos.org/build/282180989) [haskell.packages.ghc928](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc928.hoogle)
  - [[🐧✅]](https://hydra.nixos.org/build/282179378) [haskell.packages.ghc947](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc947.hoogle)
  - [[🐧✅]](https://hydra.nixos.org/build/282180135) [haskell.packages.ghc948](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc948.hoogle)
  - [[🐧✅]](https://hydra.nixos.org/build/282179670) [haskell.packages.ghc963](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc963.hoogle)
  - [[🐧✅]](https://hydra.nixos.org/build/282175474) [haskell.packages.ghc964](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc964.hoogle)
  - [[🐧✅]](https://hydra.nixos.org/build/282177719) [haskell.packages.ghc965](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc965.hoogle)
  - [[🐧✅]](https://hydra.nixos.org/build/282179627) [haskell.packages.ghc966](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc966.hoogle)
  - [[🐧✅]](https://hydra.nixos.org/build/282178863) [haskell.packages.ghc981](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc981.hoogle)
  - [[🐧✅]](https://hydra.nixos.org/build/282175394) [haskell.packages.ghc982](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc982.hoogle)
  - [[🐧✅]](https://hydra.nixos.org/build/282180308) [haskell.packages.ghc983](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc983.hoogle)
  - [[🐧✅]](https://hydra.nixos.org/build/282174639) [haskell.packages.ghc984](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc984.hoogle)
  - [[🐧✅]](https://hydra.nixos.org/build/282176201) [haskellPackages](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.hoogle)
- [ ] [[🐧❗]](https://hydra.nixos.org/build/282178363) [haskellPackages.numeric-optimization](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.numeric-optimization)  ⤴️ 1 | 2
- [ ] [hercules-ci-agent](https://hydra.nixos.org/eval/1810539?filter=hercules-ci-agent)  ⤴️ 1 | 1
  - [[🐧❗]](https://hydra.nixos.org/build/282216426) [toplevel](https://hydra.nixos.org/eval/1810539?filter=hercules-ci-agent)
  - [[🐧❗]](https://hydra.nixos.org/build/282216405) [haskellPackages](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.hercules-ci-agent)
- [ ] [[🐧❗]](https://hydra.nixos.org/build/282180662) [haskellPackages.yaml-light](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.yaml-light)  ⤴️ 0 | 5
- [ ] [[🐧❗]](https://hydra.nixos.org/build/282178430) [haskellPackages.gi-cogl](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.gi-cogl)  ⤴️ 0 | 2
- [ ] [ghc-lib](https://hydra.nixos.org/eval/1810539?filter=ghc-lib) 
  - [[🐧✅]](https://hydra.nixos.org/build/282179690) [haskell.packages.ghc8107](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc8107.ghc-lib)
  - [[🐧✅]](https://hydra.nixos.org/build/282174480) [haskell.packages.ghc902](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc902.ghc-lib)
  - [[🐧✅]](https://hydra.nixos.org/build/282175927) [haskell.packages.ghc925](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc925.ghc-lib)
  - [[🐧✅]](https://hydra.nixos.org/build/282179189) [haskell.packages.ghc926](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc926.ghc-lib)
  - [[🐧✅]](https://hydra.nixos.org/build/282176985) [haskell.packages.ghc927](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc927.ghc-lib)
  - [[🐧✅]](https://hydra.nixos.org/build/282176352) [haskell.packages.ghc928](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc928.ghc-lib)
  - [[🐧✅]](https://hydra.nixos.org/build/282178653) [haskell.packages.ghc947](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc947.ghc-lib)
  - [[🐧✅]](https://hydra.nixos.org/build/282177318) [haskell.packages.ghc948](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc948.ghc-lib)
  - [[🐧✅]](https://hydra.nixos.org/build/282181343) [haskell.packages.ghc963](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc963.ghc-lib)
  - [[🐧✅]](https://hydra.nixos.org/build/282179929) [haskell.packages.ghc964](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc964.ghc-lib)
  - [[🐧✅]](https://hydra.nixos.org/build/282181144) [haskell.packages.ghc965](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc965.ghc-lib)
  - [[🐧✅]](https://hydra.nixos.org/build/282174566) [haskell.packages.ghc966](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc966.ghc-lib)
  - [[🐧✅]](https://hydra.nixos.org/build/282179811) [haskell.packages.ghc981](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc981.ghc-lib)
  - [[🐧✅]](https://hydra.nixos.org/build/282177995) [haskell.packages.ghc982](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc982.ghc-lib)
  - [[🐧✅]](https://hydra.nixos.org/build/282179858) [haskell.packages.ghc983](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc983.ghc-lib)
  - [[🐧❗]](https://hydra.nixos.org/build/282175495) [haskell.packages.ghc984](https://hydra.nixos.org/eval/1810539?filter=haskell.packages.ghc984.ghc-lib)
  - [[🐧✅]](https://hydra.nixos.org/build/282177441) [haskellPackages](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.ghc-lib)
- [ ] [[🐧❗]](https://hydra.nixos.org/build/282177107) [haskellPackages.hcwiid](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.hcwiid) 
- [ ] [hello](https://hydra.nixos.org/eval/1810539?filter=hello) 
  - [[🐧✅]](https://hydra.nixos.org/build/282176503) [haskellPackages](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.hello)
  - [[🐧❗]](https://hydra.nixos.org/build/282216419) [pkgsCross.ghcjs.haskell.packages.ghc98](https://hydra.nixos.org/eval/1810539?filter=pkgsCross.ghcjs.haskell.packages.ghc98.hello)
  - [[🐧❗]](https://hydra.nixos.org/build/282216401) [pkgsCross.ghcjs.haskell.packages.ghcHEAD](https://hydra.nixos.org/eval/1810539?filter=pkgsCross.ghcjs.haskell.packages.ghcHEAD.hello)
  - [[🐧❗]](https://hydra.nixos.org/build/282216432) [pkgsCross.ghcjs.haskellPackages](https://hydra.nixos.org/eval/1810539?filter=pkgsCross.ghcjs.haskellPackages.hello)
  - [[🐧✅]](https://hydra.nixos.org/build/282176136) [pkgsMusl.haskellPackages](https://hydra.nixos.org/eval/1810539?filter=pkgsMusl.haskellPackages.hello)
  - [[🐧✅]](https://hydra.nixos.org/build/282180176) [pkgsStatic.haskell.packages.native-bignum.ghc948](https://hydra.nixos.org/eval/1810539?filter=pkgsStatic.haskell.packages.native-bignum.ghc948.hello)
  - [[🐧✅]](https://hydra.nixos.org/build/282175393) [pkgsStatic.haskell.packages.native-bignum.ghc984](https://hydra.nixos.org/eval/1810539?filter=pkgsStatic.haskell.packages.native-bignum.ghc984.hello)
  - [[🐧✅]](https://hydra.nixos.org/build/282174707) [pkgsStatic.haskellPackages](https://hydra.nixos.org/eval/1810539?filter=pkgsStatic.haskellPackages.hello)
- [ ] [[🐧❗]](https://hydra.nixos.org/build/282257591) [haskellPackages.hgdal](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.hgdal) 
- [ ] [[🐧❗]](https://hydra.nixos.org/build/282174604) [haskellPackages.intel-powermon](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.intel-powermon) 
- [ ] [[🐧❗]](https://hydra.nixos.org/build/282179909) [haskellPackages.mplayer-spot](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.mplayer-spot) 
- [ ] [[🐧❗]](https://hydra.nixos.org/build/282179969) [haskellPackages.numeric-optimization-ad](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.numeric-optimization-ad) 
- [ ] [[🐧❗]](https://hydra.nixos.org/build/282178952) [haskellPackages.tasty-papi](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.tasty-papi) 
- [ ] [[🐧❗]](https://hydra.nixos.org/build/282181310) [haskellPackages.termbox-banana](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.termbox-banana) 
- [ ] [[🐧❗]](https://hydra.nixos.org/build/282176839) [haskellPackages.termbox-tea](https://hydra.nixos.org/eval/1810539?filter=haskellPackages.termbox-tea) 
</details>

#### Top 50 broken packages, sorted by number of reverse dependencies
<details><summary>50 job(s) </summary>

[gogol-core](https://packdeps.haskellers.com/reverse/gogol-core) ⤴️ 184  
[haskell98](https://packdeps.haskellers.com/reverse/haskell98) ⤴️ 152  
[failure](https://packdeps.haskellers.com/reverse/failure) ⤴️ 72  
[enumerator](https://packdeps.haskellers.com/reverse/enumerator) ⤴️ 56  
[connection](https://packdeps.haskellers.com/reverse/connection) ⤴️ 53  
[util](https://packdeps.haskellers.com/reverse/util) ⤴️ 49  
[derive](https://packdeps.haskellers.com/reverse/derive) ⤴️ 48  
[web-routes](https://packdeps.haskellers.com/reverse/web-routes) ⤴️ 43  
[accelerate](https://packdeps.haskellers.com/reverse/accelerate) ⤴️ 42  
[syb-with-class](https://packdeps.haskellers.com/reverse/syb-with-class) ⤴️ 42  
[MonadCatchIO-transformers](https://packdeps.haskellers.com/reverse/MonadCatchIO-transformers) ⤴️ 41  
[TypeCompose](https://packdeps.haskellers.com/reverse/TypeCompose) ⤴️ 41  
[PrimitiveArray](https://packdeps.haskellers.com/reverse/PrimitiveArray) ⤴️ 35  
[crypto-random](https://packdeps.haskellers.com/reverse/crypto-random) ⤴️ 35  
[rank1dynamic](https://packdeps.haskellers.com/reverse/rank1dynamic) ⤴️ 33  
[dual](https://packdeps.haskellers.com/reverse/dual) ⤴️ 32  
[hsp](https://packdeps.haskellers.com/reverse/hsp) ⤴️ 32  
[distributed-static](https://packdeps.haskellers.com/reverse/distributed-static) ⤴️ 31  
[language-ecmascript](https://packdeps.haskellers.com/reverse/language-ecmascript) ⤴️ 31  
[distributed-process](https://packdeps.haskellers.com/reverse/distributed-process) ⤴️ 30  
[iteratee](https://packdeps.haskellers.com/reverse/iteratee) ⤴️ 29  
[polysemy-time](https://packdeps.haskellers.com/reverse/polysemy-time) ⤴️ 29  
[composite-base](https://packdeps.haskellers.com/reverse/composite-base) ⤴️ 28  
[polysemy-resume](https://packdeps.haskellers.com/reverse/polysemy-resume) ⤴️ 28  
[polysemy-conc](https://packdeps.haskellers.com/reverse/polysemy-conc) ⤴️ 27  
[regexpr](https://packdeps.haskellers.com/reverse/regexpr) ⤴️ 27  
[crypto-numbers](https://packdeps.haskellers.com/reverse/crypto-numbers) ⤴️ 25  
[either-unwrap](https://packdeps.haskellers.com/reverse/either-unwrap) ⤴️ 25  
[polysemy-log](https://packdeps.haskellers.com/reverse/polysemy-log) ⤴️ 25  
[HList](https://packdeps.haskellers.com/reverse/HList) ⤴️ 24  
[web-routes-th](https://packdeps.haskellers.com/reverse/web-routes-th) ⤴️ 24  
[Crypto](https://packdeps.haskellers.com/reverse/Crypto) ⤴️ 22  
[crypto-pubkey](https://packdeps.haskellers.com/reverse/crypto-pubkey) ⤴️ 22  
[haskelldb](https://packdeps.haskellers.com/reverse/haskelldb) ⤴️ 22  
[wxdirect](https://packdeps.haskellers.com/reverse/wxdirect) ⤴️ 22  
[BiobaseTypes](https://packdeps.haskellers.com/reverse/BiobaseTypes) ⤴️ 21  
[alg](https://packdeps.haskellers.com/reverse/alg) ⤴️ 21  
[mmsyn2](https://packdeps.haskellers.com/reverse/mmsyn2) ⤴️ 21  
[userid](https://packdeps.haskellers.com/reverse/userid) ⤴️ 21  
[wxc](https://packdeps.haskellers.com/reverse/wxc) ⤴️ 21  
[biocore](https://packdeps.haskellers.com/reverse/biocore) ⤴️ 20  
[reform](https://packdeps.haskellers.com/reverse/reform) ⤴️ 20  
[wxcore](https://packdeps.haskellers.com/reverse/wxcore) ⤴️ 20  
[attoparsec-enumerator](https://packdeps.haskellers.com/reverse/attoparsec-enumerator) ⤴️ 19  
[bytestring-show](https://packdeps.haskellers.com/reverse/bytestring-show) ⤴️ 19  
[cprng-aes](https://packdeps.haskellers.com/reverse/cprng-aes) ⤴️ 19  
[fay](https://packdeps.haskellers.com/reverse/fay) ⤴️ 19  
[harp](https://packdeps.haskellers.com/reverse/harp) ⤴️ 19  
[hsx2hs](https://packdeps.haskellers.com/reverse/hsx2hs) ⤴️ 19  
[incipit](https://packdeps.haskellers.com/reverse/incipit) ⤴️ 19  
</details>


*⤴️: The number of packages that depend (directly or indirectly) on this package (if any). If two numbers are shown the first (lower) number considers only packages which currently have enabled hydra jobs, i.e. are not marked broken. The second (higher) number considers all packages.*

*Report generated with [maintainers/scripts/haskell/hydra-report.hs](https://github.com/NixOS/nixpkgs/blob/haskell-updates/maintainers/scripts/haskell/hydra-report.hs)*

sternenseemann committed:
Merge branch staging-next into haskell-updates

sternenseemann committed:
jacinda: fix build by providing happy >= 2.1

This requires a few adjustments for happy's test suite due to
https://github.com/haskell/happy/commit/10a0fe5ca52d76b1e592978c95d19f899e62bee7#diff-36b847ebf9c22d0651fe4856d7a8acb47ee2c483c79f1ccf0ed8c84889c9a394

sternenseemann commented:
- [ ] darcs: <https://hydra.nixos.org/build/282180674/nixlog/9>

sternenseemann committed:
stack: fix build by providing pantry >= 0.10

Consistent usage of lself prevents inconsistent dependencies.

sternenseemann committed:
git-annex: update sha256 for 10.20241031

sternenseemann committed:
darcs: build with tls < 2.0 again

Upstream tries to do something clever with restricting tls
directly (even though it is an indirect dependency). We could easily
provide crypton-connection >= 0.4 and tls >= 2.0.6, but it only wants
that for GHC >= 9.8 claiming it is not possible for earlier versions.

sternenseemann committed:
haskellPackages.espial: allow esqueleto >= 3.5.12

sternenseemann committed:
haskellPackages.relocant: add postgresql db to tests

sternenseemann committed:
haskellPackages.HsSyck: implicit function decl may not fail build

This is another gcc 14 workaround.

haskellPackages.HsOpenSSL: don't silence incompatible-pointer-types

Disabling the error is a better option than silencing the warning which
still applies…

sternenseemann committed:
Merge branch staging-next into haskell-updates

sternenseemann committed:
haskell.packages.ghc910.http2: build with http-semantics >= 0.3

sternenseemann committed:
haskellPackages.{hd5-lite,bindings-libcddb}: disable gcc 14 -Werror

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskellPackages.termbox-bindings-c: disable implicit fun -Werror

Work around gcc 14 making implicit-function-declaration an error.

sternenseemann committed:
haskellPackages.libxml-sax: disable gcc14 implicit fun decl -Werror

sternenseemann committed:
haskellPackages: link upstream issues for GCC 14 workarounds

I've contacted the maintainer of libxml-sax directly as the package does
not seem to have a public issue tracker.

sternenseemann committed:
haskellPackages.cabal2nix-unstable: 2024-12-04 -> 2024-12-22

Includes https://github.com/NixOS/cabal2nix/pull/638.

drupol commented:
Mass ping, please resubmit a new PR.

sternenseemann commented:
This PR is like staging(-next) currently, cost of doing business at the moment unfortunately. It'll get better (#361143), but it's an unfixable situation until staging-next gets merged.

ozkutuk committed:
haskellPackages.unbound-generics: Unmark broken

vcunat commented:
That happened now.

sternenseemann commented:
Many darwin failures due to https://github.com/NixOS/nixpkgs/issues/367686, also present in master now.

sternenseemann committed:
Merge branch master into haskell-updates

sternenseemann committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh based on
*evaluation [1810632](https://hydra.nixos.org/eval/1810632) of nixpkgs commit [b3692b3](https://github.com/NixOS/nixpkgs/commits/b3692b3250716532c7bc91860f396f1e00982942) as of 2024-12-23 17:37 UTC*
from the haskell-updates jobset on hydra under https://hydra.nixos.org/jobset/nixpkgs/haskell-updates

sternenseemann committed:
haskellPackages.diagrams-input: downgrade for Stackage LTS compat

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskellPackages.amqp-streamly: don't attempt to run docker tests

sternenseemann committed:
haskellPackages: mark failing builds as broken

sternenseemann commented:
Merging with the following outstanding issues:

- GHC 9.12 packages barely work. This is also an issue on master and we'll chip away at it in the future, but we don't need to block this PR on it. Will need a new release of e.g. jailbreak-cabal: https://github.com/NixOS/jailbreak-cabal/pull/25
- GHC 9.8.4's HLS is broken. This will be solved by the next Hackage bump. Also an issue on master.
- Darwin is broken for GHC < 9.6. Tracked at #367686, also an issue on master.
- tasty-papi / mplayer-spot fail due to broken non-Haskell dependencies.

vcunat commented:
I guess I cancel the jobset?  It's still generating new jobs...

sternenseemann commented:
Yeah, those jobs won’t be shared with any other jobset. I’ll start the next iteration in the next few days.
